src/graph/graph-ops.ts

```diff
@@ -436,3 +437,25 @@ export const mapInitialEdges = (tables: Table[]): Edge[] =>
 
       return [...acc, ...items]
     }, [])
+
+const columnsWithTypeData = (table: Table): Column[] => {
+  if (table.dp_result === undefined) {
+    return table.columns
+  }
+
+  const dpColumns = table.dp_result.columns
+  const columns = table.columns.map(column => {
+    const dpColumn = dpColumns.find(dpCol => dpCol.name === column.name)
+    if (!dpColumn) {
+      return column
+    }
+
+    const columnType = {
+      dataType: dpColumn['data_type'],
+      nullable: dpColumn.nullable || false,
+    }
+    return { ...column, columnType }
+  })
+
+  return columns
+}
```

Comment:
Is this func needed? As I understand it transforms data_type to dataType. Can we use it as is without transformation?

Reply:
This function looks up for some more column data (column types) from `dp_results` matching columns by name.
It is believed the original data structure will change in the future so we either don't have to look it up in profiler or we will make lookups by `column.id` instead


Would use prefer this instead?

```
   const columnType = {
      data_type: dpColumn['data_type'],
      nullable: dpColumn.nullable || false,
    }
```